pkg/framework/proxies.go

```diff
@@ -0,0 +1,222 @@
+acl Safe_ports port 1025-65535	# unregistered ports
+acl Safe_ports port 280		# http-mgmt
+acl Safe_ports port 488		# gss-http
+acl Safe_ports port 591		# filemaker
+acl Safe_ports port 777		# multiling http
+acl CONNECT method CONNECT
+http_access deny !Safe_ports
+http_access deny CONNECT !SSL_ports
+http_access allow localhost manager
+http_access deny manager
+acl mysubnet src 192.168.0.0/16
+http_access allow localhost localnet crcnet
+http_access allow all
+http_port 80
+coredump_dir /var/spool/squid
+refresh_pattern ^ftp:		1440	20%	10080
+refresh_pattern ^gopher:	1440	0%	1440
+refresh_pattern -i (/cgi-bin/|\?) 0	0%	0
+refresh_pattern (Release|Packages(.gz)*)$      0       20%     2880
+refresh_pattern .		0	20%	4320`
+
+// CreateProxyMachineSet creates a machineset that results in a single node being spun up. This node will not be in the worker MCP.
+func CreateProxyMachineSet(client runtimeclient.Client) (*mapiv1beta1.MachineSet, error) {
+	machineSetParams := BuildMachineSetParams(client, 1)
+	machineSetParams.Labels["proxy-host"] = ""
+	machineSetParams.Labels["node-role.kubernetes.io/proxy"] = ""
+	machineSet, err := CreateMachineSet(client, machineSetParams)
+	if err != nil {
+		return nil, err
+	}
+	WaitForMachineSet(client, machineSet.GetName())
+
+	nodes, err := GetNodesFromMachineSet(client, machineSet)
+	if len(nodes) == 0 {
+		return nil, errors.New("no nodes found in machineset")
+	}
+	node := nodes[0]
+	delete(node.Labels, "node-role.kubernetes.io/worker")
+	err = client.Update(context.Background(), node)
+	return machineSet, err
+}
+
+// DeployClusterProxy Deploys an HTTP proxy to the proxy node
+func DeployClusterProxy(c runtimeclient.Client) (*appsv1.Deployment, error) {
+	squidDeploymentLabels := map[string]string{
+		"app": "squid",
+	}
+	configMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-conf",
+			Namespace: "default",
+		},
+		Data: map[string]string{
+			"squid.conf": squidConf,
+		},
+	}
+
+	err := c.Create(context.Background(), &configMap)
+	if err != nil {
+		return nil, err
+	}
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-proxy",
+			Namespace: "default",
+			Labels:    squidDeploymentLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: squidDeploymentLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "squid-proxy",
+					Namespace: "default",
+					Labels:    squidDeploymentLabels,
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "conf",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "squid-conf",
+									},
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "squid",
+							Image: "sameersbn/squid:3.5.27-2",
```

Comment:
i have one suggestion and one question here,

i think we should use the full pull spec for the image, just to ensure there are no questions of provenance.

i looked at this image and repo it is generated from, is this a common community image for squid or is there something with more recent maintenance we could use? (i'm just thinking about this test in the long term, how can we maintain this implementation)

Reply:
ah, that is a great point.  I'll need to think on that.

Reply:
I needed to restructure the test due to a restriction in host networking between nodes which I thought I had taken in to account.  During the process of addressing that, I went ahead and changed the proxy image to be built on ubi-8 and pull from registry.redhat.io.